effect.cpp

```diff
@@ -435,6 +435,41 @@ string Effects::AssembledMinion::getDescription(const ContentFactory* f) const {
   return "Can be assembled to a " + getName(f);
 }
 
+string Effects::AddAutomatonParts::getPartsNames(const ContentFactory* f) const {
+  string ret = "";
+  bool first = true;
+
+  for(ItemType part_type : part_types) {
+    if(!first) {
+      ret += ", ";
+    }
+    ret += part_type.get(f)->getName();
+    first = false;
+  }
+  return ret;
+}
+
+bool Effects::AddAutomatonParts::applyToCreature(Creature* c, Creature* attacker) const {
+  CHECK(part_types.size() > 0);
+
+  if(c->getSpareAutomatonSlots() < part_types.size()) {
```

Comment:
What's the logic behind having multiple parts in one effect and attaching all or none? Wouldn't it be simpler to have one effect -> one part?

Reply:
The logic is in exactly that, for example "automaton classses" that require all parts attached - and if that's impossible don't attach any